client/src/components/continuousLegend/index.js

```diff
@@ -110,24 +110,7 @@ const continuous = (selectorId, colorscale, colorAccessor) => {
   colors: state.colors,
 }))
 class ContinuousLegend extends React.Component {
-  constructor(props) {
-    super(props);
-    this.ref = null;
-    this.state = {
-      colorAccessor: null,
-      colorScale: null,
-    };
-  }
-
-  componentDidMount() {
-    this.updateState(null);
-  }
-
-  componentDidUpdate(prevProps) {
-    this.updateState(prevProps);
-  }
-
-  async updateState(prevProps) {
+  async componentDidUpdate(prevProps) {
```

Comment:
I see that we no longer fetch the colors on DidMount, and only rely on DidUpdate. Does that mean we don't need to render the legend between DidMount and the first DidUpdate call?

Reply:
Yes, the legend is one of the first components to be rendered(it's top level in `App.js`), and cellxgene initializes with `colorBy` set to undefined.  We won't need to render via D3 on DidMount

Reply:
Got it! Thank you 😄  SGTM!